.github/workflows/README.md

```diff
@@ -99,14 +99,67 @@ graph TD
+## Knobs on the Release Workflow
 
 One-off releases can be made with the
 [`release.yml` workflow](https://github.com/opendp/opendp/actions/workflows/release.yml)
 on github, or with the `gh` command line tool. Parameters:
 
 - **Target channel** controls how the release is tagged, and what semantic version is given to the release. There is a git branch with the same name for each channel.
-- The **sync the Channel from upstream?** checkbox is for when you want to update the `nightly`, `beta` or `stable` branches.
+- The **sync the Channel from upstream?** checkbox is for when you want to update the `nightly`, `beta` `stable` or `dev` branches. This replaces the contents of the respective branch with the branch the workflow is run from ("Use workflow from"). In practice, you'll almost always enable this.
 - Update the **version counter** accordingly when you want to release multiple nightlies or betas in the same day.
 - **Dry runs** get sent to test-pypi, and don't update the docs
 - **Fake** is for developer convenience when debugging CI: it skips compilation and inserts dummy binaries instead
+
+## Making a release
+
+1. Update `VERSION` file on the `main` branch to match the version you want to release.
+    - If it's a patch release, no change to the version number should be needed: The patch should have been incremented after the previous release.
+    - If it's a minor release, run `python tools/channel_tool.py bump_version --position minor`
+    - Similarly, if it's a major release, run `python tools/channel_tool.py bump_version --position major`
+
+1. Update changelog.
+
+    ```shell
+    python tools/channel_tool.py changelog
+    ```
+
+    Edit the contents manually. PR [#1672](https://github.com/opendp/opendp/pull/1672) provides a script to get you started (TODO: update these instructions once it is merged).
+
+1. We use a release train with nightly, beta and stable channels. Repeatedly run the Release workflow through each channel (see preceding section).
+
+    1. Wait until `nightly` is updated. It is also possible to update it yourself:
+        * Use workflow from: `Branch: main`
+        * Target Channel: `nightly`
+        * Sync the channel from Upstream?: `yes`
+
+    1. Once `nightly` is updated, update `beta`.
+        * Use workflow from: `Branch: nightly`
+        * Target Channel: `beta`
+        * Sync the channel from Upstream?: `yes`
+
+    1. Once `beta` is updated, update `stable`.
+        * Use workflow from: `Branch: beta`
+        * Target Channel: `stable`
+        * Sync the channel from Upstream?: `yes`
+
+    If CI fails, debug on main and restart.
+    If beta release failed, don't forget to increment the beta version counter next time.
+    If stable release failed, consider incrementing patch version and starting over.
+
+1. Increment patch version in `main`.
+
+    ```shell
+    python tools/channel_tool.py bump_version --position patch
+    ```
+    Add arguments as needed to bump or set version.
```

Comment:
```suggestion
```

At this point in the process, aren't we always going to only bump the patch version? The comment makes it sound like there are cases where we might do something different.